lib/fluent/plugin/buffer.rb

```diff
@@ -55,7 +55,11 @@ class BufferChunkOverflowError < BufferError; end # A record size is larger than
       # if chunk size (or records) is 95% or more after #write, then that chunk will be enqueued
       config_param :chunk_full_threshold, :float, default: DEFAULT_CHUNK_FULL_THRESHOLD
 
-      Metadata = Struct.new(:timekey, :tag, :variables)
+      Metadata = Struct.new(:timekey, :tag, :variables) do
+        def nil?
```

Comment:
Ruby's document says that "Only the object nil responds true to nil?".
This method breaks it. Change this method name to something else, like `empty?`.